components/setting_picture.tsx

```diff
@@ -66,8 +78,8 @@ export default class SettingPicture extends Component {
         }
     }
 
-    componentDidUpdate(prevProps) {
-        if (prevProps.file !== this.props.file) {
+    componentDidUpdate(prevProps: Props) {
+        if (prevProps.file !== this.props.file && this.props.file) {
```

Comment:
nit : easier to check undefined before comparision
```suggestion
        if (this.props.file && prevProps.file !== this.props.file) {
```